tools/SampleAnalyzer/Commons/DataFormat/MCSampleFormat.h

```diff
@@ -63,182 +59,243 @@ namespace MA5
-        const WeightDefinition& weight_definition() const
-        { return weight_definition_; }
+        const WeightDefinition &weight_definition() const
+        {
+            return weight_definition_;
+        }
 
         /// Accessor to the weight definition
-        WeightDefinition& weight_definition()
-        { return weight_definition_; }
+        WeightDefinition &weight_definition()
+        {
+            return weight_definition_;
+        }
 
         /// Set the PDG ID of the intial partons
         void setBeamPDGID(MAint32 a, MAint32 b)
-        {beamPDGID_=std::make_pair(a,b); }
+        {
+            beamPDGID_ = std::make_pair(a, b);
+        }
 
         /// Set the beam energy
         void setBeamE(MAfloat64 a, MAfloat64 b)
-        {beamE_=std::make_pair(a,b); }
+        {
+            beamE_ = std::make_pair(a, b);
+        }
 
         /// Set the PDF authors
         void setBeamPDFauthor(MAuint32 a, MAuint32 b)
-        {beamPDFauthor_=std::make_pair(a,b); }
+        {
+            beamPDFauthor_ = std::make_pair(a, b);
+        }
 
         /// Set the the PDF identity
         void setBeamPDFid(MAuint32 a, MAuint32 b)
-        {beamPDFID_=std::make_pair(a,b); }
+        {
+            beamPDFID_ = std::make_pair(a, b);
+        }
 
         /// Set the weight mode
         void setWeightMode(MAint32 v)
-        {weightMode_=v;}
+        {
+            weightMode_ = v;
+        }
 
         /// Set the cross section mean
         // BENJ: the normalization in the pythia lhe output by madgraph has been changed
         //       the 1e9 factor is not needed anymore
         void setXsection(MAfloat64 value)
-//  { xsection_=value*getXsectionUnitFactor();}
-        { xsection_=value;}
+        //  { xsection_=value*getXsectionUnitFactor();}
+        {
+            xsection_ = value;
+        }
 
         /// Set the cross section mean
         void setXsectionMean(MAfloat64 value)
-        { xsection_=value;}
+        {
+            xsection_ = value;
+        }
 
         /// Set the cross section mean
         void setXsectionError(MAfloat64 value)
-        { xsection_error_=value;}
+        {
+            xsection_error_ = value;
+        }
+
+        /// @brief set weight names
+        /// @param id location of the weight
+        /// @param name name of the weight
+        void SetWeightName(int id, std::string name) { weight_names_[id] = name; }
+
+        /// @brief accessor to weight names
+        const std::map<int, std::string> WeightNames() const { return weight_names_; }
 
         /// Adding a weight
         void addWeightedEvents(MAfloat64 weight)
-        { if (weight>=0) sumweight_positive_ += std::abs(weight);
-            else sumweight_negative_ += std::abs(weight); }
+        {
+            if (weight >= 0)
+                sumweight_positive_ += std::abs(weight);
+            else
+                sumweight_negative_ += std::abs(weight);
+        }
 
```

Comment:
The method `addWeightedEvents` does not seem to be used anymore. Should we get rid of it?

Reply:
I think this writes total weights in the main SAF file as a summary, we can extend it with weight collections

Reply:
I agree with the proposed extension